packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -1886,7 +2254,195 @@ class _TrackTapGestureRecognizer extends TapGestureRecognizer {
+  const ScrollbarButtonStyles({
+    this.location = ScrollbarButtonLocation.split,
+    this.extent = 48.0,
+    this.indicatorWidth = 8.0,
+    this.indicatorPaintingStyle = PaintingStyle.stroke,
+    this.leadingButtonColors = const ScrollbarButtonColors(),
+    this.trailingButtonColors = const ScrollbarButtonColors(),
+  });
+
+  /// The location of the scrollbar button.
+  final ScrollbarButtonLocation location;
+
+  /// The extent of the scrollbar button.
+  final double extent;
+
+  /// The width of the button indicator paint area.
+  final double indicatorWidth;
+
+  /// The painting style of the button indicator
+  final PaintingStyle indicatorPaintingStyle;
+
+  /// The color style of the leading button.
+  final ScrollbarButtonColors leadingButtonColors;
+
+  /// The color style of the trailing button.
+  final ScrollbarButtonColors trailingButtonColors;
+
+  /// Get the leading or trailing button indicator's path.
+  ///
+  /// By default, a simple arrow path is returned.
+  Path getIndicatorPath({
+    required Rect buttonRect,
+    required bool isLeading,
+    required ScrollbarOrientation orientation,
+  }) {
+    final Rect indicatorRect;
+    switch(orientation) {
+      case ScrollbarOrientation.left:
+      case ScrollbarOrientation.right:
+        indicatorRect = Rect.fromCenter(
+          center: buttonRect.center,
+          width: indicatorWidth,
+          height: indicatorWidth / 2.0,
+        ).intersect(buttonRect);
+        if (isLeading) {
+          return Path()
+            ..moveTo(indicatorRect.bottomLeft.dx, indicatorRect.bottomLeft.dy)
+            ..lineTo(indicatorRect.topCenter.dx, indicatorRect.topCenter.dy)
+            ..lineTo(indicatorRect.bottomRight.dx, indicatorRect.bottomRight.dy);
+        } else {
+          return Path()
+            ..moveTo(indicatorRect.topLeft.dx, indicatorRect.topLeft.dy)
+            ..lineTo(indicatorRect.bottomCenter.dx, indicatorRect.bottomCenter.dy)
+            ..lineTo(indicatorRect.topRight.dx, indicatorRect.topRight.dy);
+        }
+      case ScrollbarOrientation.top:
+      case ScrollbarOrientation.bottom:
+        indicatorRect = Rect.fromCenter(
+          center: buttonRect.center,
+          width: indicatorWidth / 2.0,
+          height: indicatorWidth,
+        ).intersect(buttonRect);
+        if (isLeading) {
+          return Path()
+            ..moveTo(indicatorRect.topRight.dx, indicatorRect.topRight.dy)
+            ..lineTo(indicatorRect.centerLeft.dx, indicatorRect.centerLeft.dy)
+            ..lineTo(indicatorRect.bottomRight.dx, indicatorRect.bottomRight.dy);
+        } else {
+          return Path()
+            ..moveTo(indicatorRect.topLeft.dx, indicatorRect.topLeft.dy)
+            ..lineTo(indicatorRect.centerRight.dx, indicatorRect.centerRight.dy)
+            ..lineTo(indicatorRect.bottomLeft.dx, indicatorRect.bottomLeft.dy);
+        }
+    }
+  }
+}
+
+/// The color styles of the button.
+class ScrollbarButtonColors {
```

Comment:
This is probably not necessary, you can probably just include foreground and background color in ScrollbarButtonStyle.